pkg/providers/vsphere/reconciler/reconciler_test.go

```diff
@@ -243,14 +244,40 @@ func TestReconcilerReconcileControlPlaneSuccess(t *testing.T) {
 	tt.Expect(tt.cluster.Status.FailureMessage).To(BeZero())
 	tt.Expect(result).To(Equal(controller.Result{}))
 
+	crs := &addonsv1.ClusterResourceSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "workload-cluster-crs-0",
+			Namespace: "eksa-system",
+		},
+	}
+	tt.ShouldEventuallyExist(tt.ctx, crs)
 	tt.ShouldEventuallyExist(tt.ctx,
-		&addonsv1.ClusterResourceSet{
+		&vspherev1.VSphereMachineTemplate{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      "workload-cluster-crs-0",
+				Name:      "workload-cluster-control-plane-1",
 				Namespace: "eksa-system",
 			},
 		},
 	)
+	capiCluster := capiCluster(func(c *clusterv1.Cluster) {
+		c.Name = "workload-cluster"
+	})
+	tt.ShouldEventuallyExist(tt.ctx, capiCluster)
+	tt.ShouldEventuallyMatch(tt.ctx,
+		crs,
+		func(g gomega.Gomega) {
+			g.Expect(crs.Spec.Resources).Should(ContainElements(addonsv1.ResourceRef{Name: "vsphere-csi-controller", Kind: "Secret"}))
```

Comment:
This is awesome, but could we do `ShouldEventuallyExist` for these secrets and confimaps too? They should be created at the same time as the CRS

Reply:
can't check on since `addonsv1.ResourceRef` doesn't have `ObjectMeta1`, that's why I use `ContainElements`

Reply:
can you do something like:
```go
    tt.ShouldEventuallyExist(tt.ctx,
		&corev1.Secret{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "vsphere-csi-controller",
				Namespace: "eksa-system",
			},
		},
	)
```